src/internal/m365/collection/drive/library_handler.go

```diff
@@ -313,18 +321,37 @@ func augmentLibraryItemInfo(
 		pps = parentPath.String()
 	}
 
-	dii.SharePoint = &details.SharePointInfo{
-		Created:    ptr.Val(item.GetCreatedDateTime()),
-		DriveID:    driveID,
-		DriveName:  driveName,
-		ItemName:   ptr.Val(item.GetName()),
-		ItemType:   details.SharePointLibrary,
-		Modified:   ptr.Val(item.GetLastModifiedDateTime()),
-		Owner:      creatorEmail,
-		ParentPath: pps,
-		SiteID:     siteID,
-		Size:       size,
-		WebURL:     weburl,
+	switch path.SharePointService {
```

Comment:
Interesting switch conditional you have there.

Kinda feels like this func should move into a different file, since it's servicing two different handlers now?  Maybe in that third file you could hook in the standard item_handler augment as well.  The only reason there were separate functions previously was because the augment funcs existed in different packages.